### PR TITLE
gh-77063: Added support for X25519 in SSLContext.set_ecdh_curve()

### PR DESCRIPTION
**Pull request by sruester**
bpo-32882: Added support for selecting X25519 in SSLContext.set_ecdh_curve()

https://bugs.python.org/issue32882

* Issue: gh-77063

### CODE REVIEW

**Comment by VeNoMouS** [comment]
Is this ever going to be implemented?

**Comment by sruester** [comment]
It is still intended to create a "unified" solution for TLS (https://www.python.org/dev/peps/pep-0543/). IMHO, until then we could well live with the "not unified, but usable" solution.

**Comment by VeNoMouS** [comment]
I ended up writing my own C API implementation that takes the ssl context ptr address

**Comment by tiran** [comment]
I have replied to you on BPO twenty minutes after you have opened the bug in 2018, https://bugs.python.org/issue32882#msg312408

tl;dr the improve is a good idea in general, but the API should be more generic and not require the user to know curve names. Therefore I proposed to use a list of curve names, preferable from an enum of supported curve names.

**Review by tiran — CHANGES_REQUESTED** [review]
See comments on BPO. New APIs should be generic and try to avoid OpenSSL-specific conventions. Let's discuss the new API on BPO first.

PS: I wasn't aware of this PR until your comments pushed it to the top of ``sort:updated-desc`` and I just happened to browse the first page. It was created before we had CODEOWNERS assignment fully configured.

**Comment by VeNoMouS** [comment]
> I have replied to you on BPO twenty minutes after you have opened the bug in 2018, https://bugs.python.org/issue32882#msg312408
> 
> tl;dr the improve is a good idea in general, but the API should be more generic and not require the user to know curve names. Therefore I proposed to use a list of curve names, preferable from an enum of supported curve names.

I disagree, the user should **also** have the ability to set curves as they set fit and not be locked down by restrictions like they currently are.

Can't we have both? have a default "preferable" list... but also give the functionality to manipulate OpenSSL with the functionality that is already available in OpensSSL and give power back to the user.

Just my 2 cents.

**Comment by tiran** [comment]
Please use bugs.python.org for discussions.
